File: epi_judge_cpp/int_as_array_increment.cc
#include <vector>

#include "test_framework/generic_test.h"
using std::vector;
vector<int> PlusOne(vector<int> A) {
  // TODO - you fill in here.

  bool append = false;
  for (auto i = A.size() - 1; i >= 0; i--) {
    if (A[i] == 9) {
      A[i] = 0;
      if (i == 0) {
        append = true;
      }
    } else {
      A[i] += 1;
      break;
    }
  }
  if (append) {
    A.insert(A.begin(), 1);
  }
  return A;
}

int main(int argc, char *argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"A"};
  return GenericTestMain(args, "int_as_array_increment.cc",
                         "int_as_array_increment.tsv", &PlusOne,
                         DefaultComparator{}, param_names);
}

File: epi_judge_python/int_as_array_increment.py
from typing import List

from test_framework import generic_test


def plus_one(A: List[int]) -> List[int]:
    if A[-1] != 9:
        A[-1] += 1
    else:
        A[-1] = 0
        residue = True
        for i in range(len(A) - 2, -1, -1):
            if residue:
                if A[i] == 9:
                    A[i] = 0
                    residue = True
                else:
                    A[i] += 1
                    residue = False
        if residue:
            A.insert(0, 1)

    return A


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('int_as_array_increment.py',
                                       'int_as_array_increment.tsv', plus_one))
